### ISSUE TITLE
UnicodeDecodeError in  sphinx.util.osutil.ustrftime (with patch)

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1128]
Problem

I'm using sphinx 1.1.3 on a German Windows 7 with python 2.7. The default encoding is cp1252.

There is a bug in sphinx.util.osutil.ustrftime. The current implementation looks like:

```
def ustrftime(format, *args):
        # strftime for unicode strings
        return time.strftime(unicode(format).encode('utf-8'), *args) \
                .decode('utf-8')
```

A call of `time.strftime('&#37;B &#37;d, &#37;Y')` returns `'M\xe4rz 13, 2013'` which is not decodable as utf-8. This causes an UnicodeDecodeError exception. The format string '&#37;B &#37;d, &#37;Y' is used by sphinx.environment.DefaultSubstitutions.apply.

The function strftime requires you to use the LC_TIME encoding. The attached patch fixes this issue.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1128
- Originally reported by: Anselm Kruis
- Originally created at: 2013-03-13T14:30:51.007

**Comment by shimizukawa** [comment]
_From Tanja Huthmacher on 2013-03-14 09:52:12+00:00_

If setlocale() has not been called before, it returns None. In this case we have to call it before giving the localEncoding string to encode().

if localEncoding is None:
            locale.setlocale(locale.LC_ALL, '')
            localEncoding = locale.getlocale(locale.LC_TIME)[1]

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-03-30 16:53:05+00:00_

Closes #1128: Fix Unicode errors when trying to format time strings with a non-standard locale.

→ <<cset 77d2a90f36a22f5e12f14dc2ef0589f122318d8d>>
